File: customer-client-app/src/app/model/customer.ts
export interface ICustomer {
    id: number,
    firstname: string,
    lastname: string,
    mobile: string,
    email: string,
    username: string,
    customerAddress: IAddress
}

export interface IAddress {
    id: number,
    addressFull: string,
    addressLine1: string,
    addressLine2: string,
    addressLine3: string,
    city: string,
    state: string,
    postcode: string,
    country: string
}

File: Domain/Customer.cs
using System;

namespace Domain
{
    public class Customer
    {
        public Guid Id { get; set; }
        public string Firstname { get; set; }
        public string Lastname { get; set; }
        public string Mobile { get; set; }
        public string Email { get; set; }
        public string Username { get; set; }
        public virtual Address CustomerAddress { get; set; }
    }
}
